internal/server/SimLoop.go

```diff
@@ -549,54 +549,18 @@ func (s *Server) ProcessAgentQueue(queuedAgents map[uuid.UUID]objects.IBaseBiker
 		// go through the agents in the queue
 		for agentNextInLineId, agentNextInLine := range queuedAgents {
 			var accepted bool
-			bikePreferenceOrder, bikeTrustHigherThanRegimeTrustMap := agentNextInLine.DecideBikePreferenceOrder()
-			nextHighestBike := s.GetMegaBikes()[bikePreferenceOrder[i]]
+			bikePreferenceOrder := agentNextInLine.DecideBikePreferenceOrder()
+			nextHighestBike := s.GetMegaBikes()[bikePreferenceOrder[i].BikeId]
```

Comment:
This line of code is used a ton. Could you add a method to the server with something like `GetBikeByID(id)`?